### DISCUSSION TITLE
Require that a Tenant is selected when adding/editing an object

### DISCUSSION THREAD

**Discussion by ziggekatten** [netbox-community/netbox#8387] [General]
Is there a way of enforcing that you always have to assign a tenant on an object using the Config Revision?

Tried some variants with o success. An example:
'''
{'tenant': {'required': 'True'}}
'''

**Comment by candlerb** [comment]
I'm not sure what you mean by "Config Revision", but have you looked at Custom Validators?

https://netbox.readthedocs.io/en/stable/customization/custom-validation/

If you're having a problem with this, then please show your complete `CUSTOM_VALIDATORS` section.

Note that `True` and `'True'` are two different things in Python (the first is a boolean, the second is a string).  See the examples [here](https://netbox.readthedocs.io/en/stable/customization/custom-validation/#plain-data).

**Comment by ziggekatten** [comment]
I'm trying to use the dynamic way as stated in the documentation using the Config Revision in the Admin section:

It does not accept  True without  ticks....
![image](https://user-images.githubusercontent.com/55629049/150146069-5a65804b-96ab-4d04-8a82-28a8dc485a95.png)

**Reply by candlerb** [reply]
I don't see any object type defined there.  Based on the documentation, I'd expect you to write:

```
{"dcim.device": [{"tenant": {"required": true}}]}
```

(assuming you wish to apply this rule to objects of type "dcim.device").  And note the square brackets: it's a *list* of validations.

**Reply by ziggekatten** [reply]
Yep, just diiscovered the fact that I missed the object type. Works like an charm when you read the docs correctly:-)

This is now solved.

**Comment by jeremystretch** [comment]
This is JSON data (not Python), so you need to use the value `True`

**Reply by candlerb** [reply]
You mean `true` ?

**Reply by jeremystretch** [reply]
Yes

**Comment by ziggekatten** [comment]
All, looking back at this, I feel documentation needs to be somewhat adjusted. From an user perspective, things are out of order. Specifying object model should come before attributes. It was not really clear. Or I'm stupid:-)

This said, how do I contribute to make documentation better? I'm happy to do so.

**Reply by jeremystretch** [reply]
Would you mind opening a [documentation change](https://github.com/netbox-community/netbox/issues/new?assignees=&labels=type%3A+documentation&template=documentation_change.yaml) citing your concerns? We'll check it out and see how the docs can be improved. Of course, you're welcome to volunteer if you'd like to put in a PR for the changes.
